src/PropertyInfo/TypeInfoParsers/PhpTypeInfoParser.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PropertyInfo\TypeInfoParsers;
+
+use PropertyInfo\Type;
+
+/**
+ *     This class will extract type information available to PHP from Properties, Getters and Setter parameters and it
+ * will parse said information into Type[] objects.
+ *
+ *     Based on the current PHP version more information will be available (ex.: PHP7 has scalar type hinting and Getter
+ * return types which are not available in PHP5).
+ *
+ * @author Mihai Stancu <stancu.t.mihai@gmail.com>
+ */
+class PhpTypeInfoParser extends NativeTypeInfoParser
+{
+    /**
+     * @param \ReflectionProperty $property
+     *
+     * @return string|null
+     */
+    public function getPropertyType(\ReflectionProperty $property)
+    {
+        return null;
+    }
+
+    /**
+     * @param \ReflectionProperty $property
+     *
+     * @return string|null
+     */
+    public function getGetterReturnType(\ReflectionProperty $property)
+    {
+        $getter = $this->getGetter($property);
+
+        if (PHP_VERSION >= 7 and $getter !== null and $getter->hasReturnType()) {
```

Comment:
`and` here too. The problem is precisely that many developers don't know the precedence trick and it can lead to bugs.
`and` and `or` are forbidden in the Symfony CS (and we follow Symfony CS) and are reported as violations by most quality checking tools including SensioLabs Insight and Scrutinizr (and we use both).